### ISSUE TITLE
Fix iOS to key loaded/unloaded off a more correct platform event

### ISSUE THREAD

**Issue by PureWeen** [dotnet/maui#13277]
### Description

Ideally, we would key loaded/unloaded off of `didMovetoWindow` on `UIView` but there's no way (AFAIK) to reach this with overriding every single view. 

The currently implementation is fiddly and probably not reliable in all scenarios. Basically, it adds observables to various layers of the view and waits until one of those triggers to fire loaded.

https://github.com/dotnet/maui/blob/main/src/Core/src/Platform/iOS/ViewExtensions.cs#L659

### Steps to Reproduce

- Run the MAUI repository
- subscribe to the loaded/unloaded event on any view
- break point the event
- see where it fires from.
- Figure out better code.

### Some thoughts
- UIWindow has methods that will tell you any time a view is added. So, we could override this method and somehow use that to notify that the window has been attached to the view. One difficulty here is that we don't have a great way to get from platform views back to xplat views. We could track all platform views we create into a ConditionalWeakTable. There has been a few cases where being able to efficiently get from platform to xplat would be useful. 
- There's been frustrations about the timing of loaded already. We could just fire `Loaded` from the first `measure` or `arrange` call on a view (though this might mean there are some cases where loaded doesn't fire and it should). We could add this on top of the current solution to make it more robust. 

### Link to public reproduction project repository

N/A

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, macOS

### Affected platform versions

ios/catalyst

**Comment by MichaelSL** [comment]
I wonder if this relates to the different behavior of `Loaded` event on Android and iOs. I have Shell application with several pages.

On Android:

1. application starts up with Page1
2. Page1 `Loaded` event handler fires
3. navigate to Page2
4. return to Page1
5. Page1 `Loaded` event doesn't fire

On iOS:

1. application starts up with Page1
2. Page1 `Loaded` event handler fires
3. navigate to Page2
4. return to Page1
5. Page1 `Loaded` event always fire

Basically, cross platform code needs workaround for iOs to avoid running same code multiple times. I wonder if this issue will help to make lifecycle events behavior more uniform?

**Comment by jstedfast** [comment]
This honestly doesn't feel like a "Good First Issue". It seems far too complicated for someone new to the team to even attempt.
